google-cloud-recaptcha_enterprise/samples/create_assessment.rb

```diff
@@ -0,0 +1,70 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START recaptcha_enterprise_create_assessment]
+
+require "google/cloud/recaptcha_enterprise/v1"
+
+def create_assessment
+  ### Create an assessment to analyze the risk of a UI action.
+
+  # Args:
+  # project_id: GCloud Project ID
+  # site_key: Site key obtained by registering a domain/app to use recaptcha services.
+  # token: The token obtained from the client on passing the recaptcha site_key.
+  # recaptcha_action: Action name corresponding to the token.
+  # assessment_name: Specify a name for this assessment.
+  ###
+
+  site_key = "your_site_key"
+  token = "user_response_token"
+  project_id = "your_project_id"
+  recaptcha_action = "action_name"
+  assessment_name = "assessment_name"
+  project_name = "projects/#{project_id}"
+
+  # Create the reCAPTCHA client.
+  client = ::Google::Cloud::RecaptchaEnterprise::V1::RecaptchaEnterpriseService::Client.new
+
+  # Set the properties of the event to be tracked.
+  event = ::Google::Cloud::RecaptchaEnterprise::V1::Event.new
+  event.site_key = site_key
+  event.token = token
+
+  assessment = ::Google::Cloud::RecaptchaEnterprise::V1::Assessment.new
+  assessment.event = event
+  assessment.name = assessment_name
+
+  # Build the assessment request.
+  request = ::Google::Cloud::RecaptchaEnterprise::V1::CreateAssessmentRequest.new
+  request.parent = project_name
+  request.assessment = assessment
```

Comment:
We prefer building requests as nested hashes rather than building proto objects themselves. Even though the types are not explicit in the code, it's a more pleasant DevX, and it's much easier to see the data structure. So for example:

```ruby
request = {
  parent: project_name,
  assessment: {
    name: assessment_name,
    event: {
      site_key: site_key,
      token: token
    }
  }
}
```